dragonfly/language/en/short_number.py

```diff
@@ -0,0 +1,103 @@
+
+'''
+ShortIntegerRef
+============================================================================
+
+:class:`ShortIntegerRef` is a modified version of :class:`IntegerRef` which allows for greater flexibility in the way that numbers may be pronounced, allowing for words like "hundred" to be dropped. This may be particularly useful when navigating files by line or page number.
+
+Some examples of allowed pronunciations:
+
+================================     ======
+Pronunciation                        Result
+================================     ======
+one                                   1
+ten                                   10
+twenty three                          23
+two three                             23
+seventy                               70
+seven zero                            70
+hundred                               100
+one oh three                          103
+hundred three                         103
+one twenty seven                      127
+one two seven                         127
+one hundred twenty seven              127
+seven hundred                         700
+thousand                              1000
+seventeen hundred                     1700
+seventeen hundred fifty three         1753
+seventeen fifty three                 1753
+one seven five three                  1753
+seventeen five three                  1753
+four thousand                         4000
+================================     ======
+
+The class works in the same way as :class:`IntegerRef`, by adding the following as an extra.
+
+.. code:: python
+
+   ShortIntegerRef("name", 0, 1000),
+'''
+
+from ..base.integer_internal  import (MapIntBuilder, CollectionIntBuilder,
+                                      MagnitudeIntBuilder, IntegerContentBase)
+from .number import int_0, int_1_9, int_10_19, int_20_90_10
+
+# Twenty five
+int_20_99       = MagnitudeIntBuilder(
+                   factor      = 10,
+                   spec        = "<multiplier> [<remainder>]",
+                   multipliers = [int_20_90_10],
+                   remainders  = [int_1_9],
+                  )
+# Two five / seven zero
+int_10_99       = MagnitudeIntBuilder(
+                   factor      = 10,
+                   spec        = "<multiplier> <remainder>",
+                   multipliers = [int_1_9],
+                   remainders  = [int_0, int_1_9],
+                  )
+# Oh five
+int_and_1_9    = CollectionIntBuilder(
+                   spec        = "(oh | zero) <element>",
```

Comment:
I think the `(oh | zero)` part should be in *number.py* instead.

Reply:
It could be, but I'm not sure this would add much. Formulations like "hundred zero five" or "five hundred oh two" are unlikely to be useful. "oh" is only really a natural formulation when you can drop the "hundred" e.g. "one oh five". At the moment in `number.py` the hundred is required.

Reply:
That is a good point. I had just noticed that `("zero", "oh")` is already present in the `DigitsContent` class at the bottom of *number.py*. I think it would be more consistent if all number content classes allowed it. It would also make zero a bit quicker to say with normal `IntegerRefs`.